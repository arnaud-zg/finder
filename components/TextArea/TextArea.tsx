import { percent, rem } from 'csx'
import { Field } from 'formik'
import React from 'react'
import { style } from 'typestyle'
import { colors } from '../../constants/colors'
import { ErrorField } from '../ErrorField'

const className = style({
  borderColor: colors.gray.toHexString(),
  borderRadius: rem(0.25),
  borderStyle: 'solid',
  borderWidth: rem(0.0625),
  fontSize: rem(1),
  marginTop: rem(0.5),
  maxHeight: rem(6),
  minHeight: rem(2),
  padding: rem(0.5),
  resize: 'vertical',
  width: percent(100),
})

interface IProps extends React.HTMLProps<HTMLInputElement> {}

export class TextArea extends React.PureComponent<IProps> {
  render() {
    return (
      <Field name={this.props.name}>
        {({ field }) => (
          <>
            <textarea
              {...field}
              {...this.props}
              className={className}
              maxlength={this.props.maxLength}
            />
            <ErrorField fieldName={field.name} />
          </>
        )}
      </Field>
    )
  }
}
