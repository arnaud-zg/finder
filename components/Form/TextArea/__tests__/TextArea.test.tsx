import { Formik } from 'formik'
import React from 'react'
import renderer from 'react-test-renderer'
import { TextArea } from '../TextArea'

describe('TextArea', () => {
  it('renders correctly', () => {
    const testRenderer = renderer.create(
      <Formik
        initialValues={{ description: '' }}
        onSubmit={() => {}}
        render={() => <TextArea name="description" />}
      />
    )
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })
})
