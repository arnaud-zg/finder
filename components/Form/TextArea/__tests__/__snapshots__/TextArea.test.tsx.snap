// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea renders correctly 1`] = `
<textarea
  className="f1tnlile"
  name="description"
  onBlur={[Function]}
  onChange={[Function]}
  value=""
/>
`;
